### ISSUE TITLE
SPIKE: adding a link to course site nav menus

### ISSUE THREAD

**Issue by klemay** [hypothesis/lms#574]
Determine whether we can add a link to the course navigation menu for the following LMSes:

- Sakai
- Moodle
- Blackboard
- D2L

We already know [this is possible in Canvas](https://canvas.instructure.com/doc/api/file.navigation_tools.html). 

If this is possible for all LMSes, providing access to activity views via a link in the nav menu could mitigate many of the complexities of developing/linking to activity views.

**Comment by klemay** [comment]
For Sakai, it looks like this would be possible if Hypothesis were added as an External Tool. See:

- https://longsight.screenstepslive.com/s/4586/m/59023/l/556701-what-are-external-tools
- https://longsight.screenstepslive.com/s/sakai_help/m/30369/l/295779-adding-the-lti-tool-to-a-single-site-via-site-info
- https://longsight.screenstepslive.com/s/4586/m/30369/l/295782-making-an-lti-tool-available-system-wide

**Comment by seanh** [comment]
I stumbled across this in the LTI spec: [Non-Context LTI Launches](https://www.imsglobal.org/specs/ltiv1p0/implementation-guide#toc-10):

> While the typical use of a Basic LTI link is in a context, it is also possible to use Basic LTI to launch a link that is not part of a context.  One example of a non-context launch might be a menu item that is part of the portal or part of a global menu in the TC [the "Tool Consumer", i.e. the LMS -- seanh].
> 
> **Supporting non-context launches is optional** for both the TP [the "Tool Provider", i.e. us -- seanh] and TC.
> 
> If a Basic LTI launch is coming from a non-context placement, the context information is simply omitted and the launch will contain the user and organization information but no context information.

"Context" in LTI spec-speak means "course", so a "non-context link" sounds like a "non-course link", so a link that's not within the context of a given assignment or a given course. That's not quite what we want -- we want a link that _is_ within the context of a specific course, just not a specific assignment.

The above quote is the entirety of what the spec has to say about this. There's nothing at all about how our app would go about creating a "non-context placement" menu item. So we'd have to rely on docs or examples from each individual LMS, like the ones above, for how it's actually done.

Basically the spec kinda obliquely hints at this functionality but isn't actually much use!

**Comment by seanh** [comment]
Hmm, actually I'm not sure that a "non-context link" is what we want here at all. We want an in-context link: the context is the course that the link to our app is placed within. But we just want that link to be placed "top-level" within the course, rather than to be specific to one assignment. I think the spec basically has no help for us here so we're relying on other docs

**Comment by seanh** [comment]
Some more docs for how adding a "course navigation" link works in Canvas, from the LTI app developer's point of view: https://canvas.instructure.com/doc/api/file.tools_xml.html

That page also explains how to make Canvas launch our app from the course menu in a new tab.

It also explains how we can add a link to our app to the "user navigation" which sounds like it might be a good place for a link to a "my activity" view.
